discussion/evaluation-of-project.tex

```diff
@@ -1,6 +1,80 @@
+
+\subsection{Code Size}
+
+From our code size analysis we have made three discoveries which are worth to discuss.
+This discussion should be read in light of the fact that the {\rust} compiler is a newly released compiler and is being compared to a {\C} compiler which have been used in production for years.
+
+\begin{description}
+
+\item [Large unoptimized {\rust} build] \hfill \\
+  As seen throughout all of our measurements in {\rust}, the unoptimized binaries are significantly larger than the optimized binaries.
+  The cause for this seems to be that the many abstractions of the {\rust} language are implemented by adding levels of indirection through function calls, like the for loop described in \autoref{sec:rust:loops}.
+  These \emph{extra} functions, which are eliminated when optimized, makes the compiler produce substantially larger binaries without the optimizations.
+
+\item [{\rust} code can be significantly larger than {\C} equivalents] \hfill \\
+  The optimized Sensor Tracker application written in {\rust} proved to be several times the size of the {\C} version.
+  This was further analyzed to reveal that the increase in size is due to the {\rust} libraries, especially considering generic code.
+  As an example, the Sensor Tracker will output samples over the \gls{usart} interface, these samples are converted to strings using the {\core} library \code{format!} macro.
+  Using this macro for the types \code{u8, i32, u32} increases the size of the optimized binary with 1736B.
+
+\item [There is zero size overhead of using {\rust}] \hfill \\
+  The Minimal Main application shows that using {\rust} by itself does not incur an increase in size.
+  It is the size of the software libraries that in our results increases the size of the {\rust} binaries
+
+\end{description}
+
+\subsection{Energy}
+
+We must emphasis that energy consumption measurements presented in \autoref{sec:res:energy} are preliminary and that the focus for the Sensor Tracker application was primarily to provide an application in this domain.
+The main problem with reading to much into these results, is that the {\rust} and {\C} implementations share a substantial amount of the library code, written in {\C}.
+The application code here is just a thin wrapper around the libraries.
+When we dug deeper into the analysis of the application we found that most of the time, the application spends its time in a busy-wait state, waiting for sensor data to be available.
+In these circumstances the difference between the two languages are not be evident.
+
+What we can take away from the results presented is that the variance in the energy consumption in the applications tested are larger within the different optimizations levels of the {\C} compiler, compared to introducing the {\rust} application layer.
```

Comment:
"results presented" -> "presented results"

"applications tested" -> "tested applications"

Evt. forslag fra leslie, siden jeg først var inne på tema om det var noen generelle regler for dette.... :P 
"From the results, we can deduce that the variance in the energy consumption in the tested applications are larger within the different levels of optimization in the C compiler, compared to the introduction of the Rust application layer. "